src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/ITaintedDataInfo.cs

```diff
@@ -14,11 +16,21 @@ internal interface ITaintedDataInfo
         /// </summary>
         string FullTypeName { get; }
 
+        /// <summary>
+        /// Qualified names of the optional dependency types.
+        /// </summary>
+        ImmutableArray<string>? DependencyFullTypeNames { get; }
```

Comment:
`ImmutableArray<T>` is a struct, so would rather not box it.

```suggestion
        ImmutableArray<string> DependencyFullTypeNames { get; }
```